src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs

```diff
@@ -69,30 +66,54 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         /// </summary>
         private class Service : IWorkspaceStatusService
         {
-            private readonly SemaphoreSlim _initializationGate = new(initialCount: 1);
             private readonly IAsyncServiceProvider2 _serviceProvider;
             private readonly IThreadingContext _threadingContext;
 
-            private bool _initialized = false;
-
             /// <summary>
             /// A task indicating that the <see cref="Guids.GlobalHubClientPackageGuid"/> package has loaded. Calls to
             /// <see cref="IServiceBroker.GetProxyAsync"/> may have a main thread dependency if the proffering package
             /// is not loaded prior to the call.
             /// </summary>
-            private JoinableTask _loadHubClientPackage;
+            private readonly JoinableTask _loadHubClientPackage;
+
+            private readonly JoinableTask<IVsOperationProgressStageStatusForSolutionLoad?> _progressStageStatus;
 
-            public event EventHandler StatusChanged;
+            public event EventHandler? StatusChanged;
 
             public Service(IAsyncServiceProvider2 serviceProvider, IThreadingContext threadingContext, IAsynchronousOperationListener listener)
             {
                 _serviceProvider = serviceProvider;
                 _threadingContext = threadingContext;
 
-                // pre-emptively make sure event is subscribed. if APIs are called before it is done, calls will be blocked
-                // until event subscription is done
-                var asyncToken = listener.BeginAsyncOperation("StatusChanged_EventSubscription");
-                Task.Run(() => EnsureInitializationAsync(CancellationToken.None), CancellationToken.None).CompletesAsyncOperation(asyncToken);
+                _loadHubClientPackage = _threadingContext.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    // Use the disposal token, since the caller's cancellation token will apply instead to the
+                    // JoinAsync operation.
+                    await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, _threadingContext.DisposalToken);
+
+                    // Make sure the HubClient package is loaded, since we rely on it for proffered OOP services
+                    var shell = await _serviceProvider.GetServiceAsync<SVsShell, IVsShell7>().ConfigureAwait(true);
+                    Assumes.Present(shell);
+
+                    await shell.LoadPackageAsync(Guids.GlobalHubClientPackageGuid);
```

Comment:
i'm not a fan of the .ConfigureAwait inconsistency.  can we either have .ConfigureAwait(true) on both of tehse, or on neither of this?

Reply:
`IVsTask` doesn't have a `ConfigureAwait` extension. Happy to remove the other one once the analyzer is updated to stop analyzing code using JTF thread switching.

Reply:
Interesting point. Feel free to file a bug on the devdiv AzDO account to add `.ConfigureAwait()` extension methods to the IVsTask type.